### PR TITLE
Error Toast when incompatible WAV resource is detected.

### PR DESCRIPTION
**Pull request by android-usb**
Also PlayStore resources.

### CODE REVIEW

File: samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt [hunk 1]
```diff
@@ -62,39 +62,61 @@ class DrumPlayer {
...
-        loadWavAsset(assetMgr, "CrashCymbal.wav", CRASHCYMBAL, PAN_CRASHCYMBAL)
-        loadWavAsset(assetMgr, "RideCymbal.wav", RIDECYMBAL, PAN_RIDECYMBAL)
-        loadWavAsset(assetMgr, "MidTom.wav", MIDTOM, PAN_MIDTOM)
-        loadWavAsset(assetMgr, "LowTom.wav", LOWTOM, PAN_LOWTOM)
-        loadWavAsset(assetMgr, "HiHat_Open.wav", HIHATOPEN, PAN_HIHATOPEN)
-        loadWavAsset(assetMgr, "HiHat_Closed.wav", HIHATCLOSED, PAN_HIHATCLOSED)
+    fun loadWavAssets(assetMgr: AssetManager): Boolean {
+        var allAssetsCorrect = true
+        if (!loadWavAsset(assetMgr, "KickDrum.wav", BASSDRUM, PAN_BASSDRUM)) {
+            allAssetsCorrect = false
```
Comment (philburk) [inline samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt:68]: allAssetsCorrect &= loadWavAssets(...) is also an option
More terse but a bit opaque.

File: samples/drumthumper/src/main/cpp/DrumPlayerJNI.cpp [hunk 2]
```diff
@@ -79,13 +80,19 @@ JNIEXPORT void JNICALL Java_com_plausiblesoftware_drumthumper_DrumPlayer_loadWav
     WavStreamReader reader(&stream);
     reader.parse();
 
+    int sampleRate = reader.getSampleRate();
+    int numChannels = reader.getNumChannels();
+    jboolean isFormatValid = sampleRate == 41000 && numChannels == 1;
```
Comment (philburk) [inline samples/drumthumper/src/main/cpp/DrumPlayerJNI.cpp:85]: Somewhere there is probably a 44100 used when opening the stream.
This code and that code should share a constant, maybe kRequiredSampleRate.
That will make it easier for people to replace the 44100 Hz samples with another rate.
Same for numChannels.

File: samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt (hunk 1 above)
Comment (android-usb) [reply samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt:68]: Wish I had thought of that! (however, Kotlin doesn't have an &&= operator, so a slightly different solution is required).

File: samples/drumthumper/src/main/cpp/DrumPlayerJNI.cpp (hunk 2 above)
Comment (android-usb) [reply samples/drumthumper/src/main/cpp/DrumPlayerJNI.cpp:85]: Ack. Done.

File: samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt [hunk 3]
```diff
@@ -104,7 +89,11 @@ class DrumPlayer {
             var dataLen = assetFD.getLength().toInt()
             var dataBytes: ByteArray = ByteArray(dataLen)
             dataStream.read(dataBytes, 0, dataLen)
+<<<<<<< HEAD
```
Comment (philburk) [inline samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt:92]: Unresolved merge conflict.

File: samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt [hunk 4]
```diff
@@ -64,30 +65,14 @@ class DrumPlayer {
...
-        if (!loadWavAsset(assetMgr, "LowTom.wav", LOWTOM, PAN_LOWTOM)) {
-            allAssetsCorrect  = false
-        }
-        if (!loadWavAsset(assetMgr, "HiHat_Open.wav", HIHATOPEN, PAN_HIHATOPEN)) {
-            allAssetsCorrect  = false
-        }
-        if (!loadWavAsset(assetMgr, "HiHat_Closed.wav", HIHATCLOSED, PAN_HIHATCLOSED)) {
-            allAssetsCorrect  = false
-        }
+        allAssetsCorrect = loadWavAsset(assetMgr, "KickDrum.wav", BASSDRUM, PAN_BASSDRUM) && allAssetsCorrect
```
Comment (philburk) [inline samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt:68]: Nice.

File: samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt [hunk 5]
```diff
@@ -22,18 +22,19 @@ import java.io.IOException
...
+        val NUM_SAMPLE_CHANNELS: Int = 1;   // All WAV resource must be mono
         val SAMPLE_RATE: Int = 44100    // All the input samples are assumed to BE 44.1K
                                         // All the input samples are assumed to be mono.
 
         // Sample Buffer IDs
         val BASSDRUM: Int = 0
         val SNAREDRUM: Int = 1
         val CRASHCYMBAL: Int = 2
-        val RIDECYMBAL: Int = 3
+        const val RIDECYMBAL: Int = 3
```
Comment (philburk) [inline samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt:37]: Why is just this one val declared const?

File: samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt [hunk 6]
```diff
@@ -113,10 +102,14 @@ class DrumPlayer {
         return returnVal
     }
 
-    external fun setupAudioStreamNative(numChannels: Int, sampleRate: Int)
+    external fun setupAudioStreamNative(sampleRate: Int, numChannels: Int)
     external fun teardownAudioStreamNative()
 
+<<<<<<< HEAD
```
Comment (philburk) [inline samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt:108]: Unresolved merge conflict.

File: samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt (hunk 4 above)
Comment (android-usb) [reply samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt:68]: Ack. Don't know how that slipped in there.

File: samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt (hunk 5 above)
Comment (android-usb) [reply samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt:37]: Hmmm..... "const" isn't even a Kotlin thing.

**Pull request merged**